sdk/typescript/src/utils/suins.ts

```diff
@@ -2,8 +2,8 @@
 // SPDX-License-Identifier: Apache-2.0
 
 const SUI_NS_NAME_REGEX =
-	/^(?:[a-z0-9][a-z0-9-]{0,62}(?:\.[a-z0-9][a-z0-9-]{0,62})*)?@[a-z0-9][a-z0-9-]{0,62}$/i;
-const SUI_NS_DOMAIN_REGEX = /^(?:[a-z0-9][a-z0-9-]{0,62}\.)+sui$/i;
+	/^(?!.*--)(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?(?:\.[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?)*)?@[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?$/i;
+const SUI_NS_DOMAIN_REGEX = /^(?!.*--)(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+sui$/i;
```

Comment:
```suggestion
const SUI_NS_DOMAIN_REGEX = /^(?!.*(^(?!@)|[-.@])($|[-.@]))(?:[a-z0-9-]{0,63}\.)+sui$/i;
```

Reply:
Updated the regexes a bit:
Instead of filtering out `--` specifically, this adds a negative lookahead at the beginning that matches `-`, `@`,  `.`, or `-`/`.` appearing next to another character in that group or at the begging/end (except for `@` at the beginning of the name variant)

Reply:
LGTM